Tests/OpenAITests/OpenAITests.swift

```diff
@@ -89,17 +89,22 @@ class OpenAITests: XCTestCase {
 
     func testToolCallStreamResponse() async throws {
         MockURLProtocol.mockNetworkHandlers[OpenAI.ChatCompletionRequest.path] = { request in
+            MockURLProtocol.mockNetworkHandlers[OpenAI.ChatCompletionRequest.path] = { request in
+                return (.success(try self.getData(filename: "tool_call_stream_response", fileExtension: "txt")!), 200)
+            }
```

Comment:
I'm interested in why this works, though I think it would be better to pass this in a `FunctionSchema` callback. This really tests the MockURLProtocol rather than the library itself.